src/main/java/com/exasol/swqa/maven/SummarizerMojo.java

```diff
@@ -0,0 +1,120 @@
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    @SuppressWarnings("unused")
+    private MavenProject project;
+
+    @Override
+    // [impl -> qs~failing-safely-when-summarization-breaks~1]
+    public void execute() {
+        try {
+            summarize();
+        } catch (final Exception exception) {
+            getLog().warn("The following issue occurred during quality metric summarization: '" + exception.getMessage()
+                    + "'. Continuing build since this step is optional.");
+        }
+    }
+
+    private void summarize() throws MojoFailureException {
+        final Path mavenTargetPath = prepareTargetDirectory();
+        final Path jacocoXMLPath = mavenTargetPath.resolve("site").resolve("jacoco.xml");
+        final float coverage = extractCoverageFromJaCoCoXML(jacocoXMLPath);
+        writeSummaryFile(mavenTargetPath, coverage);
+    }
+
+    private Path prepareTargetDirectory() {
+        return Path.of(project.getBuild().getDirectory());
+    }
+
+    // [impl -> dsn~extracting-code-coverage-from-ja-co-co-report~1]
+    private float extractCoverageFromJaCoCoXML(final Path jacocoXMLPath) throws MojoFailureException {
+        try {
+            final Document document = getXMLDocument(jacocoXMLPath);
+            final NodeList reportNodes = document.getElementsByTagName("report");
+            final Element reportNode = (Element) reportNodes.item(0);
+            final NodeList counterNodes = reportNode.getElementsByTagName("counter");
+            for (int i = 0; i < counterNodes.getLength(); i++) {
+                final Element counterNode = (Element) counterNodes.item(i);
+                if ("BRANCH".equals(counterNode.getAttribute("type"))) {
+                    return calculateBranchCoverage(counterNode);
+                }
+            }
+            return 0.0f;
+        } catch (final IOException | ParserConfigurationException | SAXException exception) {
+            throw new MojoFailureException("Failed to extract coverage from the JaCoCo XML: " + jacocoXMLPath,
+                    exception);
+        }
+    }
+
+    private float calculateBranchCoverage(Element counterNode) {
+        final int missedBranches = Integer.parseInt(counterNode.getAttribute("missed"));
+        final int coveredBranches = Integer.parseInt(counterNode.getAttribute("covered"));
+        final int allBranches = missedBranches + coveredBranches;
+        float branchCoveragePercentage = coveredBranches * 100.0f / allBranches;
+        getLog().debug("Branch coverage is " + branchCoveragePercentage + ". " + coveredBranches + " of " + allBranches
+                + " covered.");
+        return branchCoveragePercentage;
+    }
+
+    private static Document getXMLDocument(Path jacocoXMLPath)
+            throws ParserConfigurationException, SAXException, IOException {
+        final DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
```

Comment:
Configure document builder factory to avoid attacks against various dangerous XML features.